pwa-devdocs/scripts/create-reference-docs/templates/renderReferenceTable.js

```diff
@@ -0,0 +1,36 @@
+const requiredProp = '<i class="material-icons green">check_box</i>';
+
+function renderEntry({ key, required, defaultValue, propDescription }) {
+    return `| \`${key}\` | ${required} | ${defaultValue} | ${propDescription}|\n`;
+}
+
+module.exports = ({ props, propsOverrides }) => {
+    let renderedContent = `
+## Props
+
+| Name | Required | Default | Description |
+| --- | :---: | :---: | --- |
+`;
+
+    let propKeys = Object.keys(props);
+
+    propKeys.forEach(key => {
+        let prop = propsOverrides
+            ? Object.assign(props[key], propsOverrides[key])
+            : props[key];
+
+        let templateValues = {
+            key: key,
+            required: prop.required ? requiredProp : '`-`',
+            defaultValue: prop.defaultValue ? `\`${prop.defaultValue.value}\`` : '`-`',
+            propDescription: prop.description?prop.description:'`-`'
+        };
+
+        renderedContent += renderEntry(templateValues);
+    });
+
+    renderedContent += '{:style="table-layout:auto"}';
+
+
+    return renderedContent;
+};
```

Comment:
Is there any style / structure we have to enforce in the comment blocks to get these reference tables to come out nicely?

Reply:
Values for Name and Required are pretty predictable and probably don't need enforcement.
However, Default and Description values have the potential to be more diverse, and I'll have to keep an eye out on how some of those values will interact with the table and adjust the template or docblock accordingly.